chromadb/utils/embedding_functions.py

```diff
@@ -751,18 +751,39 @@ def __init__(
     def __call__(self, input: Documents) -> Embeddings:
         accept = "application/json"
         content_type = "application/json"
-        embeddings = []
-        for text in input:
-            input_body = {"inputText": text}
+        provider = self._model_name.split('.')[0]
+        if provider == "amazon":
+            embeddings = []
+            for text in input:
+                input_body = {
+                    "inputText": text
+                }
+                body = json.dumps(input_body)
+                response = self._client.invoke_model(
+                    body=body,
+                    modelId=self._model_name,
+                    accept=accept,
+                    contentType=content_type,
+                )
+                embedding = json.load(response.get("body")).get("embedding")
+                embeddings.append(embedding)
+        elif provider == "cohere":
+            # See Amazon Bedrock User Guide > Cohere Embed models for more information
+            # https://docs.aws.amazon.com/bedrock/latest/userguide/model-parameters-embed.html
+            input_body = {
+                "texts": input,
+                "input_type": "search_document"
+            }
             body = json.dumps(input_body)
             response = self._client.invoke_model(
                 body=body,
                 modelId=self._model_name,
                 accept=accept,
                 contentType=content_type,
             )
-            embedding = json.load(response.get("body")).get("embedding")
-            embeddings.append(embedding)
+            embeddings = json.load(response.get("body")).get("embeddings")
+        else:
```

Comment:
could we fast-fail at construction instead? and output a useful error to the user, so they know which models are available?

Reply:
Since the list of models and model versions are constantly changing but the payload stays mostly the same, how about a fast-fail based on model provider?
![image](https://github.com/chroma-core/chroma/assets/7282984/db3fe48b-9c5f-4b7c-a53c-aeb9aa1d35f3)

Reply:
Let's do that.

Reply:
Implemented